pkg/nsx/services/subnetport/builder.go

```diff
@@ -102,3 +110,61 @@ func getObjectMeta(obj interface{}) *metav1.ObjectMeta {
 func getCluster(service *SubnetPortService) string {
 	return service.NSXConfig.Cluster
 }
+
+func buildSubnetPortExternalAddressBindingFromExisting(subnetPort *model.VpcSubnetPort, existingSubnetPort *model.VpcSubnetPort) *model.VpcSubnetPort {
+	if existingSubnetPort == nil {
+		return subnetPort
+	}
+	if existingSubnetPort.ExternalAddressBinding != nil {
+		if subnetPort.ExternalAddressBinding != nil {
+			// update is not supported
+			subnetPort.ExternalAddressBinding = &model.ExternalAddressBinding{}
+		}
+	}
+	return subnetPort
+}
+
+func (service *SubnetPortService) buildExternalAddressBinding(sp *v1alpha1.SubnetPort) *model.ExternalAddressBinding {
+	if service.GetAddressBindingBySubnetPort(sp) != nil {
+		return &model.ExternalAddressBinding{}
```

Comment:
Where you set the external address binding with a non-empty value?

Reply:
This line will add an empty ExternalAddressBinding into SubnetPort, and then NSX will allocate ExternalAddressBinding or keep existing one. Set this value to nil will delete ExternalAddressBinding in SubnetPort.